### DISCUSSION TITLE
SERIAL COMMUNICATION

### DISCUSSION THREAD

**Discussion by Flyer133** [CrashOverride85/zc95#72] [General]
Hey CrashOverride85!

Is it possible to send simple commands via serial line:   Arduino--> Zc95.
For example, I went to use a board ([example board](https://fr.aliexpress.com/item/1005003153036762.html?spm=a2g0o.order_list.order_list_main.41.d61d5e5bXEmH9m&gatewayAdapt=glo2fra)) to control multiple relays and many sensors.

A simple language would be a must, for example:
A_PowerUp x --> Increase the power to A line to x
A_PowerDown x --> Decrease the power to A line to x
ProgSet x --> Set The prog number x
Stop --> Stop all channels
...

Thanks ;-)
Flyer133

**Comment by CrashOverride85** [comment]
Hi Flyer133,
sort of - but it's not simple atm. The Python GUI can be used over serial, and that same protocol could be implemented from an Arduino.

I'll see if I can add an Arduino library that talks to the zc95 over serial in the next couple of weeks or so

**Comment by CrashOverride85** [comment]
I've added a basic Arduino library - although you can't do a great deal with it yet:
https://github.com/CrashOverride85/zc95/tree/main/misc/Arduino/libraries/Zc95

**Reply by Flyer133** [reply]
Thanx a lot CrashOverride85!!!
This code will be very useful!!

**Comment by RupertB-ES** [comment]
Can BLE Streaming be done over the aux port too ? or is it just going to be via BLE

**Comment by CrashOverride85** [comment]
right now, no, the streaming support is only for BLE. I have been thinking about adapting it for use over serial, it would need to work a little differently though
